### ISSUE TITLE
Release 4.9.0 - Beta 2 - E2E UX tests - Amazon Security Hub

### ISSUE THREAD

**Issue by juliamagan** [wazuh/wazuh#25167]
## End-to-End (E2E) Testing Guideline

- **Documentation:** Always consult the development documentation for the current stage tag at [this link](https://documentation-dev.wazuh.com/v4.9.0-beta2/index.html). Be careful because some of the description steps might refer to a current version in production, always navigate using the current development documention for the stage under test. Also, visit the following [pre-release package guide](https://docs.google.com/document/d/1fW8koy8P8OMvcxzIOXWGDw9qmKlju-XuRHD-Bdv3uP4/edit) to understand how to modify certain links and urls for the correct testing of the development packages.
- **Test Requirements:** Ensure your test comprehensively includes a full stack and agent/s deployment as per the Deployment requirements, detailing the machine OS, installed version, and revision.
- **Deployment Options:** While deployments can be local (using VMs, Vagrant, etc) or on the aws-dev account, opt for local deployments when feasible. For AWS access, coordinate with the DevOps team through [this link](https://github.com/wazuh/internal-devel-requests/issues/new?assignees=&labels=level%2Ftask%2C+request%2Foperational%2C+type%2Fchange&projects=&template=operational--request.md&title=%3CTitle%3E).
- **External Accounts:** If tests require third-party accounts (e.g., GitHub, Azure, AWS, GCP), request the necessary access through the DevOps team [here](https://github.com/wazuh/internal-devel-requests/issues/new?assignees=&labels=level%2Ftask%2C+request%2Foperational%2C+type%2Fchange&projects=&template=operational--request.md&title=%3CTitle%3E).
- **Alerts:** Every test should generate a minimum of one end-to-end alert, from the agent to the dashboard, irrespective of test type.
- **Multi-node Testing:** For multi-node wazuh-manager tests, ensure agents are connected to both workers and the master node.
- **Package Verification:** Use the pre-release package that matches the current TAG you're testing. Confirm its version and revision.
- **Filebeat Errors:** If you encounter errors with Filebeat during testing, refer to [this Slack discussion](https://wazuh-team.slack.com/archives/C03BDG0K6JC/p1672168163537809) for insights and resolutions.
- **Known Issues:** Familiarize yourself with previously reported issues in the Known Issues section. This helps in identifying already recognized errors during testing.
- **Reporting New Issues:** Any new errors discovered during testing that aren't listed under Known Issues should be reported. Assign the issue to the corresponding team (QA if unsure), add the `Release testing` objective and `Urgent` priority. Communicate these to the team and QA via the c-release Slack channel.
- **Test Conduct:** It's imperative to be thorough in your testing, offering enough detail for reviewers. Incomplete tests might necessitate a redo.
- **Documentation Feedback:** Encountering documentation gaps, unclear guidelines, or anything that disrupts the testing or UX? Open an issue, especially if it's not listed under Known Issues. Please answer the feedback section, this is a mandatory step.
- **Format:** If this is your first time doing this, refer to the format (but not necessarily the content, as it may vary) of previous E2E tests, here you have an example https://github.com/wazuh/wazuh/issues/13994.
- **Status and completion:** Change the issue status within your team project accordingly. Once you finish testing and write the conclusions, move it to Pending review and notify the @wazuh/devel-pyserver team via Slack using the [c-release channel](https://wazuh-team.slack.com/archives/C02A737S5MJ). Beware that the reviewers might request additional information or task repetitions.
- **For reviewers:** Please move the issue to Pending final review and notify via Slack using the same thread if everything is ok, otherwise, perform an issue update with the requested changes and move it to On hold, increase the review_cycles in the  team project by one and notify the issue assignee via Slack using the same thread.

For the conclusions and the issue testing and updates, use the following legend:

**Status legend**
- 🟢 All checks passed
- 🟡 Found a known issue
- 🔴 Found a new error

## Issue delivery and completion
- **Initial delivery:** The issue's assignee must complete the testing and deliver the results by Aug 12, 2024 and notify the @wazuh/devel-pyserver team via Slack using the [c-release channel](https://wazuh-team.slack.com/archives/C02A737S5MJ)
- **Review:** The @wazuh/devel-pyserver team will assign a reviewer and add it to the `review_assignee` field in the  project. The reviewer must then review the test steps and results. Ensure that all iteration cycles are completed by Aug 13, 2024 date (issue must be in `Pending final review` status) and notify the QA team via Slack using the [c-release channel](https://wazuh-team.slack.com/archives/C02A737S5MJ).
- **Auditor:** The QA team must audit, validate the results, and close the issue by Aug 14, 2024.

## Deployment requirements

| Component | Installation | Type | OS |
|----------|--------------|------|----|
| Indexer | [Quickstart](https://documentation-dev.wazuh.com/v4.9.0-beta2/quickstart.html) | Single Node | Ubuntu 22.04 x86_64 |
| Server | Same as Indexer, all-in-one | Single node | Same as Indexer |
| Dashboard | Same as indexer, all-in-one | - | Same as Indexer |
| Agent | [Installing Wazuh agents](https://documentation-dev.wazuh.com/v4.9.0-beta2/installation-guide/wazuh-agent/index.html) | - | Debian 12 x86_64 |

## Test description

- Configure AWS Security Hub in a Wazuh Manager and a Wazuh Agent. Follow the documentation page  https://documentation-dev.wazuh.com/v4.9.0-beta2/cloud-security/amazon/services/supported-services/security-hub.html
- Generate Amazon Security Hub alerts. Check that generated alerts appear on the dashboard.
- Use [discard_regex option](https://documentation-dev.wazuh.com/v4.9.0-beta2/user-manual/reference/ossec-conf/wodle-s3.html#bucket-discard-regex) to effectively discard certain types of events. 

## Known issues

There are no known issues.

## Test conclusions

Status | Test | Failure type | Notes
-- | -- | -- | --
🟡    | Configure AWS Security Hub. Follow the documentation page https://documentation-dev.wazuh.com/v4.9.0-beta2/cloud-security/amazon/services/supported-services/security-hub.html | The documentation was complex and I could not replicate the process by myself | Known issue: https://github.com/wazuh/wazuh-documentation/issues/7588
🟢    | Configure AWS Security Hub in the Wazuh server. | I successfully configured the AWS Security Hub on my Wazuh server without encountering any errors | 
🟢    | Configure AWS Security Hub in the Wazuh agent. | I successfully configured the AWS Security Hub on my Wazuh agent without encountering any errors | This should indicate that the previous issue has been fixed https://github.com/wazuh/wazuh-documentation/issues/7566
🟡  | Generate Amazon Security Hub alerts. Check that generated alerts appear on the dashboard. | The alerts did not appear in the Amazon Web Services dashboard. They only appeared on the **Discover** dashboard | The Amazon Web Service dashboard should support the AWS Security Hub integration.
🟢    | Use [discard_regex option](https://documentation-dev.wazuh.com/v4.9.0-beta2/user-manual/reference/ossec-conf/wodle-s3.html#bucket-discard-regex) to effectively discard certain types of events. | The events that include the specified field were discarded. Please see https://github.com/wazuh/wazuh/issues/25167#issuecomment-2290885954

## Feedback

We value your feedback. Please provide insights on your testing experience.

- Was the testing guideline clear? Were there any ambiguities?
  - The test objectives were properly spelled out, so I had no challenges understanding what needed to be done.

- Did you face any challenges not covered by the guideline?
  - I faced lots of challenges replicating the process within the guidelines. The **documentation** needs to be worked on.

- Suggestions for improvement:
  - I would advise that the process be explained to someone from the Content team for proper rework
  - I advise the integration to include coverage by the **Amazon Web Services dashboard**.

## Reviewers validation

The criteria for completing this task are based on the validation of the conclusions and the test results by all reviewers.

All the checkboxes below must be marked in order to close this issue.
- [ ] @davidjiglesias
- [ ] @wazuh/devel-pyserver

**Comment by Ifex370** [comment]
## Environment install :green_circle: 

<img width="640" alt="image" src="https://github.com/user-attachments/assets/5626fb12-3b0b-4a7f-b743-1fb8f2637cee">

### Wazuh Server install :green_circle:

#### Resources:
VirtualBox VM, 8GB RAM, 2 cores, 50Gb disk
Ubuntu 22.04 fresh install, fully upgraded

<details><summary>Prerequisites</summary>

**Comment by Ifex370** [comment]
The [AWS Security Hub documentation](https://documentation-dev.wazuh.com/v4.9.0-beta2/cloud-security/amazon/services/supported-services/security-hub.html) requires root privileges to complete the requirements. I utilized my environment and tried to follow the documentation.

This documentation is still quite new and still requires a lot of optimization in other to boost clarity. However, I followed the process as best as I could.

## AWS Configuration ⚫

### Prerequisite: [Configure AWS Config](https://docs.aws.amazon.com/securityhub/latest/userguide/securityhub-setup-prereqs.html#securityhub-prereq-config) 

For you to enable the AWS security hub, you need to be a delegated administrator. The documentation presents 2 different ways to configure Security Hub, and I utilized the Manual method which was straight-forward since I am the administrator :green_circle: 

<img width="1905" alt="image" src="https://github.com/user-attachments/assets/72899009-94ca-4a15-8113-130d9bf5077c">

### AWS managed policies attachment to the IAM identity :green_circle: 

The documentation mentions two policies that should be attached to the user profile created:
- AWSSecurityHubFullAccess
- AWSSecurityHubOrganizationsAccess

<img width="1904" alt="image" src="https://github.com/user-attachments/assets/dffa18df-48da-42a1-be55-492cb7d6a1fd">

<img width="1900" alt="image" src="https://github.com/user-attachments/assets/762bddf0-96f3-44cc-827f-9498aa3a9634">

### Integrating Security Hub with EventBridge
According to the documentation, you need to create the following resources to integrate Security Hub with EventBridge:

#### 1. A Firehose stream in Amazon Data Firehose. 
- It gives the guide [Amazon WAF integration](https://documentation-dev.wazuh.com/v4.9.0-beta2/cloud-security/amazon/services/supported-services/waf.html) for a Firehose configuration example.

#### 2. An event rule in EventBridge

I created the event rule using a SecurityHub Custom action:
<img width="1901" alt="image" src="https://github.com/user-attachments/assets/ced26c3b-abd9-4305-ac65-a8b4848fe18f">

The documentation further details two external AWS documentations that should help in guiding the user for the creation of event rules on the AWS EventBridge

- [Creating an event rule for automatically sent findings](https://docs.aws.amazon.com/securityhub/latest/userguide/securityhub-cwe-all-findings.html#securityhub-cwe-all-findings-predefined-pattern)

- [Defining a rule for using custom actions to send findings and insight results](https://docs.aws.amazon.com/securityhub/latest/userguide/securityhub-cwe-custom-actions.html#securityhub-cwe-define-rule)

I assume these steps are not necessary since we already have a custom action created using the first guide provided in this section: create a [custom action in Security Hub](https://docs.aws.amazon.com/securityhub/latest/userguide/securityhub-cwe-custom-actions.html).

**Comment by Ifex370** [comment]
## Test conclusions

Status | Test | Failure type | Notes
-- | -- | -- | --
🔴   | Configure AWS Security Hub. Follow the documentation page https://documentation-dev.wazuh.com/v4.9.0-beta2/cloud-security/amazon/services/supported-services/security-hub.html | The documentation was complex and I could not replicate the process by myself | Known issue: https://github.com/wazuh/wazuh-documentation/issues/7588
🟢    | Configure AWS Security Hub in the Wazuh server. | I successfully configured the AWS Security Hub on my Wazuh server without encountering any errors | 
🟢    | Configure AWS Security Hub in the Wazuh agent. | I successfully configured the AWS Security Hub on my Wazuh agent without encountering any errors | This should indicate that the previous issue has been fixed https://github.com/wazuh/wazuh-documentation/issues/7566
🟡  | Generate Amazon Security Hub alerts. Check that generated alerts appear on the dashboard. | The alerts did not appear in the Amazon Web Services dashboard. They only appeared on the **Discover** dashboard | The Amazon Web Service dashboard should support the AWS Security Hub integration.
🟡 | Use [discard_regex option](https://documentation-dev.wazuh.com/v4.9.0-beta2/user-manual/reference/ossec-conf/wodle-s3.html#bucket-discard-regex) to effectively discard certain types of events. | In progress

**Comment by fdalmaup** [comment]
# Review 

The `discard_regex` option test case is yet to be completed. To achieve it, the output of the execution of the module should be attached to the [issue comment](https://github.com/wazuh/wazuh/issues/25167#issuecomment-2288593975) and the dashboard should not show the discarded logs.

The logs will look like the ones found in the [development issue](https://github.com/wazuh/wazuh/issues/21166#issuecomment-1971968855):
```
DEBUG: +++ The "Testing.*" regex found a match in the "actionName" field. The event will be skipped.
DEBUG: +++ The "Testing.*" regex found a match in the "actionName" field. The event will be skipped.
DEBUG: +++ The "Testing.*" regex found a match in the "actionName" field. The event will be skipped.
```

Putting the issue On Hold until this is achieved.

**Comment by fdalmaup** [comment]
Hi @Ifex370. You could check the logs that you got in the Wazuh Dashboard to define the regex and the field to be discarded, or, as in the [development issue](https://github.com/wazuh/wazuh/issues/21166#issuecomment-1971968855), use the value of the Custom Action that is triggered for events of type `Security Hub Findings - Custom Action`. To filter the logs from the `ossec.log` you can use the following command:
`cat /var/ossec/logs/ossec.log | grep "aws\|^DEBUG"`
It will output the logs containing `aws` or beginning with `DEBUG`.

**Comment by Ifex370** [comment]
I have configured the discard_regex option to:

```
 <wodle name="aws-s3">
      <disabled>no</disabled>
      <interval>5m</interval>
      <run_on_start>yes</run_on_start>
      <subscriber type="security_hub">
          <sqs_name>security-hub-rules-testing</sqs_name>
          <aws_profile>prod</aws_profile>
          <discard_regex field="finding.Resources.Region">us-east-1</discard_regex>
      </subscriber>
  </wodle>
```

I would need a new Security Hub event to trigger this configuration.

**Comment by fdalmaup** [comment]
# Review

The `discard_regex` parameter was correctly used and the logs and screenshot shared show that no log was processed since these were skipped.

LGTM!
